### PR TITLE
Added Abiphone contact requirements and location.

### PR DESCRIPTION
**Pull request by hannibal002**
Added 2 features around the abiphone missing contacts inventory:

- Abiphone NPC Requirements - Show what the NPC needs in order to add him as contact in the abiphone
- Abiphone NPC Location - Click on an NPC to set an marker at the location

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/AbiphoneContactHelper.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+		JsonObject data = getJsonData(npcName);
+		if (data == null) return;
+		if (!data.has("x")) return;
+
+		int x = data.get("x").getAsInt();
+		int y = data.get("y").getAsInt();
+		int z = data.get("z").getAsInt();
+
+		String islandName = fetchIslandName(stack);
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/AbiphoneContactHelper.java:121]: even if this works i would probably prefer having the island in the repo

Comment (jani270) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/AbiphoneContactHelper.java:121]: effort

Comment (hannibal002) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/AbiphoneContactHelper.java:121]: It works and as jani noted, it's more effort for him. Or is your argument about performance? We can cache the whole lore

Comment (jani270) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/AbiphoneContactHelper.java:121]: > It works and as jani noted, it's more effort for him. Or is your argument about performance? We can cache the whole lore

I added the island in my draft repo pr, if you want to you can change it in code

**Comment by jani270** [comment]
> Added 2 features around the abiphone missing contacts inventory:
> 
> * Abiphone NPC Requirements - Show what the NPC needs in order to add him as contact in the abiphone
> * Abiphone NPC Location - Click on an NPC to set an marker at the location

Needed branch: https://github.com/jani270/NotEnoughUpdates-REPO/tree/broo

**Review by NopoTheGamer — DISMISSED** [review]

**Review by romangraef — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**